Addon infrastructure

simonihmig opened the pull request:
This overhauls the addon infrastructure:
* Use latest Ember CLI setup
* Add first simple test
* Move component to addon folder, to support Babel transpilation
* Some code style updates (ES6)

This sets the groundwork for another PR yet to follow...

simonihmig commented:
@ef4 seems Travis is not picking this up automatically, and needs to be enabled manually?

dcyriller commented on .travis.yml:
```diff
@@ -0,0 +1,67 @@
+node_js:
+  # we recommend testing addons with the same minimum supported node version as Ember CLI
+  # so that your addon works for all apps
+  - "6"
```
Node 8 is probably what you'd like

simonihmig replied:
Yes, thought about it, but decided to stick with the default addon blueprint of the latest Ember CLI for now, and to not make this necessarily a breaking change.

But I am fine with making that change, either as part of this PR or a separate one!?

Also related to breaking changes: I didn't find any notice about supported Ember versions. As this adds the default ember-try scenarios testing back to 2.18, this would be the minimum supported version (or do we want to be more aggressive?). We should make this explicit in the Readme, and would have to make this a breaking change release anyway, right? 

Please tell me how you want to handle this, and I'll make the necessary updates.

dcyriller replied:
> Yes, thought about it, but decided to stick with the default addon blueprint of the latest Ember CLI for now, and to not make this necessarily a breaking change.

They have been updated only a few days ago 😬.

> But I am fine with making that change, either as part of this PR or a separate one!?

This one is fine IMHO

> and to not make this necessarily a breaking change.

Node 8 would be breaking... But Node 6 as well. Currently, there is no minimum Node version set (neither in package.json nor in CI).

I don't feel comfortable with using Node 6 as it's not maintained by the Node team anymore. And there are chances that we don't update this very soon...

dcyriller replied:
> We should make this explicit in the Readme,

We might want to add this to the README (taken from the blueprints):

```md
Compatibility
------------------------------------------------------------------------------

* Ember.js v3.4 or above
* Ember CLI v2.13 or above
* Node.js v8 or above
```

dcyriller commented on config/ember-try.js:
```diff
@@ -0,0 +1,84 @@
+      useYarn: true,
+      scenarios: [
+        {
+          name: 'ember-lts-2.18',
```
Could be replaced by `ember-lts-3.8`?

simonihmig replied:
Is it already an LTS? Also see the other comment on supported Ember versions, then I can update this if required!

dcyriller replied:
According to this doc https://emberjs.com/releases/, `3.8` is already LTS. Digging a bit, the blueprint update has been merged only 5 days ago 😬. I think this scenario should be added. Sorry for the manual update I'm asking you to do.

IMHO, there is no point in supporting ember versions not supported by ember anymore. (Even though they most probably work.) So we should drop 2.18 support. But @ef4 may disagree.

ef4 replied:
There is a lot of value in keeping support for older Ember versions. For one thing, it makes it easier for people to do upgrades, because they can upgrade addons like this one first, and then upgrade Ember itself.

It's fine to drop older Ember versions if they're hard to support, but if the addon already works under that version anyway, there's no reason not to support and test it.

dcyriller commented:
Thank you for that @simonihmig! I think it's a great update.

I can't help with the Travis setup though.

ef4 commented:
Thanks @simonihmig, this is great. I enabled Travis.

ef4 commented:
I think it's fine that you used the current version of the blueprints, no reason to try to rush ahead and get changes that are only on ember-cli master.

ef4 commented:
Ember CLI 3.10 itself officially doesn't work on Node 6, which is why tests are failing. Let's use the same `engines` statement that Ember CLI 3.10 uses.

simonihmig commented:
So after both of your feedback, I
* kept the Ember scenarios as-is, i.e. Ember 2.18 is the minimum (though we *could* support much earlier versions, as we don't rely on any specific new features AFAICT)
* dropped Node 6
* added "Compatibility" and the other Readme sections from the addon blueprint
* fixed the linting config and some dependencies (uncovered through the linter)

Turbo87 commented:
FWIW dropping support for Node 6 should have been considered a breaking change, requiring a new major version release. This is currently causing the CI for ember-cp-validations to fail :-/

ef4 commented:
Ah, true. Sorry.

On Thu, Sep 26, 2019 at 9:14 AM Tobias Bieniek <notifications@github.com>
wrote:

> FWIW dropping support for Node 6 should have been considered a breaking
> change, requiring a new major version release. This is currently causing
> the CI for ember-cp-validations to fail :-/
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/ef4/ember-code-snippet/pull/60?email_source=notifications&email_token=AACN6MVJHA4KZMX36TPTWPTQLSYUBA5CNFSM4HPU73K2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7VQORQ#issuecomment-535496518>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACN6MWDXGITYRHLEYQW7K3QLSYUBANCNFSM4HPU73KQ>
> .
>

jelhan commented:
@ef4 Also having trouble with that one while working on a backport on Ember Bootstrap. Did you consider releasing a `v2.4.2`, which reverts the breaking change?

ef4 commented:
Re-published 2.4.0 as 2.4.2 to address the breaking change.
